lua/autorun/sh_waiter.moon

```diff
@@ -0,0 +1,82 @@
+require "moonscript"
+
+AddCSLuaFile!
+
+export Waiter
+export WaiterQueue
+
+Waiter = {}
+WaiterQueue = WaiterQueue or {}
+
+Waiter.lastLoopDuration = 0
+
+patronQueue = {}
+patronCount = 0
+
+-- Minimum delay (in seconds) between executions of attendPatrons()
+minBreakAfterAttending = 0.5
+
+generatePatron = (waitingFor, onSuccess, onTimeout, maxAttempts=0) ->
+    print "[CFC Waiter] Generating new Patron!"
+    patron =
+        :onSuccess,
+        :onTimeout,
+        :waitingFor,
+        :maxAttempts
+        attempts: 0
+
+    patronQueue[patronCount] = patron
+
+    patronCount += 1
```

Comment:
```suggestion
class PatronCount
    new: =>
        @count = 0
    increment: =>
        @count++
        
     patronCount = PatronCount!
     patronCount.increment()
```

;)

But really, why don't we `class` this bic tf up

Reply:
I think we want everyone to use the same Waiter "instance" just given the way waiter is set up

Reply:
That there's called a `Singleton` class